Use HTTPS URL schema everywhere

ypid opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note
    that our contribution policy requires that a feature request or bug
    report be opened for approval prior to filing a pull request. This
    helps avoid wasting time and effort on something that we might not
    be able to accept.

    Please indicate the relevant feature request or bug report below.
    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ACCEPTED BUG REPORT OR
    FEATURE REQUEST, IT WILL BE MARKED AS INVALID AND CLOSED.
-->
### Fixes: #5504
<!--
    Please include a summary of the proposed changes below.
-->

Mechanical edit which replaces `http://` by `https://` for URLs where it is known that the web server supports HTTPS.

I basically wanted to get ride of http://netbox.readthedocs.io/ in the UI (Docs link) :wink:

ypid committed:
Use HTTPS everywhere (mechanical edit using util from https-everywhere)

```Shell
node ~/src/EFForg/https-everywhere/utils/rewriter/rewriter.js .
git checkout netbox/project-static/
```

A few additional changes where reset manually before the commit.

jeremystretch reviewed (changes requested):
There is no reason to modify test data. Please revise your PR so that changes are restricted only to the project's documentation.

ypid committed:
Use HTTPS everywhere (mechanical edit using util from opening_hours.js)

```Shell
make -f ~/src/opening-hours/opening_hours.js/Makefile qa-https-everywhere
git checkout netbox/project-static/
git checkout netbox/*/tests
```

ypid commented:
The idea with the tests was to just default to HTTPS. Anyway, I updated the patchset.
